astropy/io/fits/card.py

```diff
@@ -959,7 +1053,98 @@ def _format_comment(self):
         else:
             return f' / {self._comment}'
 
-    def _format_image(self):
+    @property
+    def _full_unit_regexp(self):
+        s = re.escape(self.unit_str_start)
+        l = re.escape(self.unit_str_delimiters[0])
+        r = re.escape(self.unit_str_delimiters[1])
+        e = re.escape(self.unit_str_end)
+        unit_str_position = self.unit_str_position.lower()
+        if unit_str_position in ['start', 'beginning']:
+            ms = '^'
+            me = ''
+        elif unit_str_position == 'end':
+            ms = ''
+            me = '$'
+        else:
+            raise ValueError(f'Unknown unit string position {self.unit_str_position}.  Expecting "start" or "end"')
+        return f'{ms}{s}{l}.*{r}{e}{me}'
+
+    def _get_comment_unit(self, comment):
+        """Returns `~astropy.units.Unit` found in comment, if defined, else ``None``"""
+        # Extract unit together with lead-in and delimiter
+        # https://stackoverflow.com/questions/8569201/get-the-string-within-brackets-in-python
+        m = re.search(self._full_unit_regexp, comment)
+        if m is None:
+            #log.debug(f'no unit matching "{self._full_unit_regexp}" in "{comment}"')
+            return None
+        # Strip off delemeters
+        punit_str = m.group(0)
+        s = re.escape(self.unit_str_start)
+        l = re.escape(self.unit_str_delimiters[0])
+        r = re.escape(self.unit_str_delimiters[1])
+        e = re.escape(self.unit_str_end)
+        unit_str = re.sub(f'[{s}{l}{r}{e}]', '', punit_str)
+        try:
+            unit = u.Unit(unit_str)
+        except ValueError as e:
+            warnings.warn(
+                f'Unit not assigned to {self.keyword} keyword because of '
+                f'Unit conversion error in comment '
+                f"'{comment}': {e}", AstropyUserWarning)
+            return None
+        return unit
+
+    def _full_unit_str(self, unit):
+        unit_str = unit.to_string()
```

Comment:
For writing, I think we should use `format='fits'` for sure, so that other fits readers have a chance.

Reply:
Thanks for the pointer to that feature